src/specklepy/objects/base.py

```diff
@@ -463,22 +463,18 @@ def add_detachable_attrs(self, names: Set[str]) -> None:
 
     @property
     def units(self) -> Union[str, None]:
-        if self._units:
-            return self._units.value
-        return None
+        return self._units
 
     @units.setter
     def units(self, value: Union[str, Units, None]):
-        if value is None:
-            units = value
+        if isinstance(value, str) or value is None:
+            self._units = value
         elif isinstance(value, Units):
-            units: Units = value
+            self._units = value.value
         else:
-            units = get_units_from_string(value)
```

Comment:
as discussed with Gergo, removing this conversion to the common string representation, might be a breaking change (in case the code was relying on always being able to detect common string representation), although the new behavior is more aligned with C#

Reply:
This is intended, and aligned with sharp.
The only breaking change will be for users of specklepy who are creating objects with mapped unit strings. e.g. "millimetres" no longer will map to "mm".

If we can make sure our first party connectors are sending correct units. I'll add a comment that geometry expects specific unit strings. This should be sufficient.